lib/rex/socket/sctp.rb

```diff
@@ -0,0 +1,66 @@
+# -*- coding: binary -*-
+require 'rex/socket'
+require 'rex/io/stream'
+
+class ::Socket
+  IPPROTO_SCTP = 132
+  SOL_SCTP     = 132
+end
```

Comment:
Thoughts on adding these constants to `::Rex::Socket` instead?

Reply:
At the end of the day, `Rex::Socket`s are still `Socket`s so i figure the constant belongs with the rest of them. No objection to moving if we have a good reason.

Reply:
The rationale behind the comment was - we shouldn't patch core files we don't own, that's why `::Rex::Socket` exists in the first place right?

There's also load order issues if a module ends up depending on this patched `::Socket` constant, but this file hasn't been loaded etc

Reply:
Kind of? Rex::Socket is really a factory for the dynamic dispatch of API-consistent "virtual wiring" but it uses ::Socket for "everything" under the skin. If you're worried about namespace corruption, just take a look at .class.ancestors in almost any MSF object - some of the gems we pull in do things that make my work look downright pristine wink
IMO, I'm not so much corrupting someone else's namespace as i'm fixing a missing element in their implementation - that's a real Socket class per the RFC, and they just neglected to define the constant.

Reply:
I'll treat this as not a blocker, but I'm not a massive fan of the pattern